backend-base/src/main/java/org/eclipse/che/incubator/workspace/telemetry/base/AbstractAnalyticsManager.java

```diff
@@ -195,51 +383,94 @@ public AbstractAnalyticsManager(String apiEndpoint, String workspaceId, String m
         LOG.warn("Cannot retrieve user Id from the machine token", e);
       }
     }
+    return userId;
   }
 
-  private Workspace getWorkspace(String endpoint) {
-    try {
-      return this.requestFactory.fromUrl(endpoint).request().asDto(WorkspaceDto.class);
-    } catch (IOException |
-        ServerException |
-        UnauthorizedException |
-        ForbiddenException |
-        NotFoundException |
-        ConflictException |
-        BadRequestException e) {
-      throw new RuntimeException("Can't get workspace information for Che analytics", e);
-    }
-
+  /**
+   * create a map of the common and current event properties merged together
+   *
+   * @return a map of the current event and common workspace properties
+   */
+  public Map<String, Object> getCurrentEventProperties(Map<String, Object> eventProperties) {
+    ImmutableMap.Builder<String, Object> currentEventPropertiesBuilder = ImmutableMap.builder();
+    commonProperties.forEach((k, v) -> {
+      currentEventPropertiesBuilder.put(k, v);
+    });
+    eventProperties.forEach((k, v) -> {
+      currentEventPropertiesBuilder.put(k, v);
+    });
+    return currentEventPropertiesBuilder.build();
   }
 
-  private List<String> getPluginNamesFromWorkspace(Workspace ws) {
-    List<String> pluginNames = new ArrayList<String>();
-    List<? extends Component> components = ws.getDevfile().getComponents();
-    return components.stream()
-      .filter((e -> e.getType().equals("chePlugin")))
-      .map((e -> e.getId()))
-      .collect(Collectors.toList());
+  private Map<String, Object> makeCommonProperties() {
+    ImmutableMap.Builder<String, Object> commonPropertiesBuilder = ImmutableMap.builder();
+
+    Arrays.asList(new SimpleImmutableEntry<>(EventProperties.CREATED, createdOn),
+        new SimpleImmutableEntry<>(EventProperties.WORKSPACE_ID, workspaceId),
+        new SimpleImmutableEntry<>(EventProperties.WORKSPACE_NAME, workspaceName),
+        new SimpleImmutableEntry<>(EventProperties.UPDATED, updatedOn),
+        new SimpleImmutableEntry<>(EventProperties.STOPPED, stoppedOn),
+        new SimpleImmutableEntry<>(EventProperties.AGE, age),
+        new SimpleImmutableEntry<>(EventProperties.RETURN_DELAY, returnDelay),
+        new SimpleImmutableEntry<>(EventProperties.FIRST_START, firstStart),
+        new SimpleImmutableEntry<>(EventProperties.STACK_ID, stackId),
+        new SimpleImmutableEntry<>(EventProperties.FACTORY_ID, factoryId),
+        new SimpleImmutableEntry<>(EventProperties.FACTORY_NAME, factoryName),
+        new SimpleImmutableEntry<>(EventProperties.FACTORY_URL, factoryUrl),
+        new SimpleImmutableEntry<>(EventProperties.FACTORY_OWNER, factoryOwner),
+        new SimpleImmutableEntry<>(EventProperties.LAST_WORKSPACE_FAILED, stoppedAbnormally),
+        new SimpleImmutableEntry<>(EventProperties.LAST_WORKSPACE_FAILURE, lastErrorMessage),
+        new SimpleImmutableEntry<>(EventProperties.OSIO_SPACE_ID, osioSpaceId),
```

Comment:
Would be nice to remove obsolete + Hosted Che specific `EventProperties.OSIO_SPACE_ID`